mlrun/db/httpdb.py

```diff
@@ -5353,6 +5346,33 @@ def _wait_for_background_task_from_response(self, response):
             )
         return None
 
+    def _resolve_page_params(self, params: typing.Optional[dict]) -> dict:
+        """
+        Resolve the page parameters, setting defaults where necessary.
+        """
+        page_params = deepcopy(params) or {}
+        if page_params.get("page-token") is None and page_params.get("page") is None:
+            page_params["page"] = 1
+        if page_params.get("page-size") is None:
+            page_size = config.httpdb.pagination.default_page_size
+
+            if page_params.get("limit") is not None:
+                page_size = params["limit"]
```

Comment:
why use `params` again?
```suggestion
            if limit := page_params.get("limit"):
                page_size = limit
```

Reply:
your suggestion wont catch limit=0

Reply:
Right. 
But off-topic - limit=0 should be a valid input at all 👀

Reply:
not the client to decide :)